misk-jooq/src/main/kotlin/misk/jooq/JooqTransacter.kt

```diff
@@ -61,11 +83,12 @@ class JooqTransacter(
   }
 
   private fun <RETURN_TYPE> createDSLContextAndCallback(
+    options: TransacterOptions,
     callback: (jooqSession: JooqSession) -> RETURN_TYPE
   ): RETURN_TYPE {
     var jooqSession: JooqSession? = null
     return try {
-      dslContext.value.transactionResult { configuration ->
+      dslContext(dataSourceService, clock, dataSourceConfig, options).transactionResult { configuration ->
```

Comment:
is this doing extra work on each transaction and is there a chance it could slow things down? one thing that looks unnecessary is creating the settlings and mappings.

Reply:
This actually hasn't changed much. Previously the creation of the DSL context was 'lazy'. The DSL context was being created every time before as well.

Reply:
we discussed this offline, looks like it is doing no additional work here.